OpenRA.Mods.Common/Traits/BotModules/UnitBuilderBotModule.cs

```diff
@@ -161,32 +147,39 @@ void BuildUnit(IBot bot, string name)
 
 		ActorInfo ChooseRandomUnitToBuild(ProductionQueue queue)
 		{
-			var buildableThings = queue.BuildableItems();
-			if (!buildableThings.Any())
+			var unitsToBuild = Info.UnitsToBuild.ToArray();
```

Comment:
Creating an array each call. Better would be to cache this in a field.

Reply:
I don't see the array being used after the line below? Why even create it?

Reply:
fixed